src/h5web/vis-packs/core/heatmap/MappedHeatmapVis.tsx

```diff
@@ -55,17 +60,12 @@ function MappedHeatmapVis(props: Props): ReactElement {
 
   const baseArray = useBaseArray(value, slicedDims);
   const dataArray = useMappedArray(baseArray, slicedMapping);
-
-  const domain = useMemo(() => {
-    return customDomain || getDomain(dataArray.data as number[], scaleType);
-  }, [customDomain, dataArray, scaleType]);
+  const dataDomain =
+    useDomain(dataArray.data as number[], scaleType) || DEFAULT_DOMAIN;
 
   useEffect(() => {
-    if (!customDomain) {
-      // If auto-scale is on, update `dataDomain`
-      setDataDomain(domain);
-    }
-  }, [customDomain, domain, setDataDomain]);
+    setDataDomain(dataDomain);
+  }, [dataDomain, setDataDomain]);
```

Comment:
Now, I always set the `dataDomain` in the config whenever it changes (i.e. when we switch dataset or scale type) -- I'm basically fixing #486, ahead of #510.